tests/export_test.py

```diff
@@ -668,6 +743,43 @@ def test_multi_platform_and_poly(self):
     res2 = export.call_exported(exp2)(x)
     self.assertAllClose(res2, _testing_multi_platform_fun_expected(x).reshape((-1,)))
 
+  def test_multi_platform_and_sharding(self):
+    nr_devices = 2
+    export_devices = jax.devices()[0:nr_devices]
+    export_mesh = Mesh(export_devices, axis_names=("x",))
+    a = np.arange(16 * 4, dtype=np.float32).reshape((16, 4))
+    @functools.partial(
+        jax.jit,
+        in_shardings=(jax.sharding.NamedSharding(export_mesh, P("x", None),),),
+        out_shardings=jax.sharding.NamedSharding(export_mesh, P(None, "x")))
+    def f_jax(b):  # b: f32[16 // DEVICES, 4]
+      return b * 2.
+
+    res_native = f_jax(a)
+    exp = export.export(f_jax,
+                        lowering_platforms=("cpu", "tpu"))(a)
+
+    expected_re = re.compile(
```

Comment:
Might be easier to just check the HLO instead of StableHLO but up to you

Reply:
The regexp will be different but in the end it is the same kind of thing. And StableHLO is what JAX actually lowers to, HLO is obtained after some post-processing.